src/extensions/default/InlineBoxModelEditor/main.js

```diff
@@ -26,48 +26,56 @@ define(function (require, exports, module) {
         ExtensionUtils = brackets.getModule("utils/ExtensionUtils"),
-        InlinePaddingEditor = require("InlinePaddingEditor").InlinePaddingEditor,
-        properties = JSON.parse(require("text!PaddingProperties.json")),
-        PaddingUtils = require("PaddingUtils"),
+        InlineBoxModelEditor = require("InlineBoxModelEditor").InlineBoxModelEditor,
+        properties = JSON.parse(require("text!BoxModelProperties.json")),
+        BoxModelUtils = require("BoxModelUtils"),
         InlineWidget = brackets.getModule("editor/InlineWidget").InlineWidget;
  
-    var DEFAULT_PADDING = "15px";
+    var DEFAULT_BoxModel = "15px";
+    var type;
 
     /**
      * editor context if so; otherwise null.
      *
      * @param {Editor} hostEditor
      * @param {{line:Number, ch:Number}} pos
-     * @return {?{paddingValue:String, marker:TextMarker}}
+     * @return {?{BoxModelValue:String, marker:TextMarker}}
      */
     function prepareEditorForProvider(hostEditor, pos) {
-        var paddingRegEx, paddingValueRegEx, cursorLine, match, sel, start, end, endPos, marker;
+        var BoxModelRegEx, BoxModelValueRegEx, cursorLine, match, sel, start, end, endPos, marker;
 
         sel = hostEditor.getSelection();
         if (sel.start.line !== sel.end.line) {
             return null;
         }
 
-        paddingRegEx = new RegExp(PaddingUtils.PADDING_REGEX);
-        paddingValueRegEx = new RegExp(PaddingUtils.PADDING_VALUE_REGEX);
+        BoxModelRegEx = new RegExp(BoxModelUtils.BoxModel_REGEX);
+        BoxModelValueRegEx = new RegExp(BoxModelUtils.BoxModel_VALUE_REGEX);
         cursorLine = hostEditor.document.getLine(pos.line);
 
-        // Loop through each match of paddingRegEx and stop when the one that contains pos is found.
+        // Loop through each match of BoxModelRegEx and stop when the one that contains pos is found.
         do {
-            match = paddingRegEx.exec(cursorLine);
+            match = BoxModelRegEx.exec(cursorLine);
             if (match) {
                 start = match.index;
                 end = start + match[0].length;
             }
         } while (match && (pos.ch < start || pos.ch > end));
 
         if(match){
-            // Check if the cursorLine has a CSS rule of type padding
-            var cssPropertyName, semiColonPos, colonPos, paddingValue, cursorLineSubstring, firstCharacterPos;
+            // Check if the cursorLine has a CSS rule of type BoxModel
+            var cssPropertyName, semiColonPos, colonPos, BoxModelValue, cursorLineSubstring, firstCharacterPos;
 
-            // Get the css property name after removing spaces and ":" so that we can check for it in the file PaddingProperties.json
+            // Get the css property name after removing spaces and ":" so that we can check for it in the file BoxModelProperties.json
             cssPropertyName = cursorLine.split(':')[0].trim();
 
+            if(cssPropertyName === "margin") {
+                type = "margin";
```

Comment:
let's not make it a global variable. declare it in this function and add it as a property in the return value. Then, use `context.type` in the function that calls this function.